File: frontend/src/types/job.ts
// Job-related types
export interface Job {
  id: number;
  job_id?: string;
  company: string;
  job_title: string;
  job_description: string;
  job_url?: string;
  location?: string;
  salary_min?: number;
  salary_max?: number;
  source: string;
  status: string;
  job_type?: string;
  match_score?: number;
  recommendation?: 'apply_now' | 'apply_with_confidence' | 'consider_carefully' | 'not_recommended' | 'skip';
  created_at: string;
  updated_at: string;
  application_deadline?: string;
  notes?: string;
  strengths?: string[];
  concerns?: string[];
  missing_skills?: string[];
}

export interface JobCreate {
  company: string;
  job_title: string;
  job_description: string;
  job_url?: string;
  location?: string;
  salary_min?: number;
  salary_max?: number;
  source: string;
}

export interface JobFilters {
  status_filter?: string;
  min_score?: number;
  source?: string;
  skip?: number;
  limit?: number;
  location?: string;
  job_type?: string;
  company?: string;
  search?: string;
}

export interface JobAnalysis {
  job_id: number;
  match_score: number;
  recommendation: string;
  strengths: string[];
  concerns: string[];
  analysis_details: {
    semantic_similarity: number;
    skill_match: number;
    location_match: number;
    salary_match: number;
  };
  created_at: string;
}

File: backend/app/schemas/job.py
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class JobBase(BaseModel):
    company: str
    job_title: str
    job_description: str
    job_url: str
    location: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    remote_type: Optional[str] = "unknown"
    job_type: Optional[str] = "full-time"
    source: str = "manual"


class JobCreate(JobBase):
    pass


class JobUpdate(BaseModel):
    company: Optional[str] = None
    job_title: Optional[str] = None
    job_description: Optional[str] = None
    status: Optional[str] = None
    match_score: Optional[float] = None
    analysis_results: Optional[Dict[str, Any]] = None
    drive_folder_id: Optional[str] = None
    drive_folder_url: Optional[str] = None
    applied_date: Optional[datetime] = None


class JobResponse(JobBase):
    id: int
    job_id: str
    match_score: Optional[float]
    analysis_completed: bool
    status: str
    drive_folder_url: Optional[str]
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class JobList(BaseModel):
    total: int
    jobs: List[JobResponse]
    skip: int
    limit: int


class JobFromExtension(BaseModel):
    jobTitle: str
    company: str
    location: Optional[str] = None
    jobDescription: str
    jobUrl: str
    salary: Optional[str] = None
    source: str


class JobProcessResponse(BaseModel):
    success: bool
    jobId: int
    matchScore: float
    message: str
    driveUrl: Optional[str] = None
    status: str
